pkg/apis/storage/types.go

```diff
@@ -341,7 +341,7 @@ type CSIDriverSpec struct {
 	// unset or false and it can be flipped later when storage
 	// capacity information has been published.
 	//
-	// This is an alpha field and only available when the CSIStorageCapacity
+	// This is a beta field and only available when the CSIStorageCapacity
```

Comment:
We have a new `+featureGate=CSIStorageCapacity` tag now!

Reply:
Does that replace this entire "This is a beta field and ..." paragraph or do I add the tag to that?

Reply:
The only other place that uses it does so without `+`:
https://github.com/kubernetes/kubernetes/commit/72da0b1bb06607f3f3e067f1bb5ce329ec861e1b#r47879733

Reply:
I've left this out because it didn't seem to make any difference in the generated files.

Reply:
It's new, we don't have a lot of examples yet, and we are not consuming it yet, but we are encouraging people to start setting it.

Reply:
There seems to be some confusion about the tag. You are saying it should be `+featureGate=CSIStorageCapacity` whereas @nikhita said in https://github.com/kubernetes/kubernetes/pull/98277#discussion_r582558737 that it should be `featureGate=CSIStorageCapacity`.

Reply:
Using plain text (as in the current PR) and then adding the tag with `+featureGate` make sense to me, though, so I don't mind adding that.

Reply:
@nikhita can you clarify the format we should use here?

Reply:
FWIW, the reason why I prefer `+featureGate` is that it does not end up in the human-readable text like the `featureGate=LoadBalancerClass` in https://github.com/kubernetes/kubernetes/blob/55a5d4faf38b8cc9f65c301b9e16278048336bac/staging/src/k8s.io/api/core/v1/types_swagger_doc_generated.go#L2260

That's what we have the full sentence for.

Reply:
I think we can trust Tim's guidance here. He's formalizing it in a KEP:
https://github.com/kubernetes/enhancements/issues/2558

Reply:
@nikhita said in https://kubernetes.slack.com/archives/C5P3FE08M/p1615188110025800?thread_ts=1615058923.018400&cid=C5P3FE08M:

> I think that should be fine because the comment lines with +featureGate will be ignored by the generators, etc now. We can update it after the KEP is finalized.

Reply:
My intention was for it to be metadata:  `+featureGate`.  If we ultimately decide that was wrong, we can always change it, but I suspect this is better

Reply:
It was added to the v1beta types.go. @thockin do you want to add it to the internal types too?

There was also question about the semantic of this tag. Does it refer to the feature gate controlling:
- whether or not values are persisted in the API server (ie we drop values if the feature is disabled)
- whether or not this api is functional (the object gets created, but nothing happens because the feature gate is turned off in the consuming components)

Reply:
It is meant to be processed when generating code for OpenAPI documentation, so I'd expect that it is only relevant in the external API.

Regarding the semantic: this tag is still WIP. I expect that sorting out such details will need further discussion during review of https://github.com/kubernetes/enhancements/pull/2560